File: Business Rule/Assemblies/Workforce Planning (WPP)/Admin Management Dashboard (AMD)/amd/sql/migrations.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Globalization;
using System.IO;
using System.Linq;
using OneStream.Finance.Database;
using OneStream.Finance.Engine;
using OneStream.Shared.Common;
using OneStream.Shared.Database;
using OneStream.Shared.Engine;
using OneStream.Shared.Wcf;
using OneStream.Stage.Database;
using OneStream.Stage.Engine;
using OneStreamWorkspacesApi;
using OneStreamWorkspacesApi.V800;

namespace Workspace.__WsNamespacePrefix.__WsAssemblyName
{
	public class Migrations
	{
		//Get all the table classes
		List<IMigration> tables = new List<IMigration> {
			new Workspace.__WsNamespacePrefix.__WsAssemblyName.component_config()
		};

		#region "Get Migration Queries"
		
		public List<string> GetMigrationQueries(SessionInfo si, string type)
		{
			try
			{
				//Declare list of queries
				List<string> queries = new List<string>();
				
				//Loop through all the tables to get the queries
				foreach (var table in tables)
				{
					queries.Add(table.GetMigrationQuery(si, type));
				}
				
				return queries;
			}
			catch (Exception ex)
			{
				throw new XFException(si, ex);
			}
		}
		
		#endregion

		#region "Get Population Queries"
		
		public List<string> GetPopulationQueries(SessionInfo si, string type)
		{
			try
			{
				//Declare list of queries
				List<string> queries = new List<string>();
				
				//Loop through all the tables to get the queries
				foreach (var table in tables)
				{
					queries.Add(table.GetPopulationQuery(si, type));
				}
				
				return queries;
			}
			catch (Exception ex)
			{
				throw new XFException(si, ex);
			}
		}
		
		#endregion

	}
}

File: Business Rule/Assemblies/RES/RES/RES/sql/migrations.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Common
Imports System.Globalization
Imports System.IO
Imports System.Linq
Imports Microsoft.VisualBasic
Imports OneStream.Finance.Database
Imports OneStream.Finance.Engine
Imports OneStream.Shared.Common
Imports OneStream.Shared.Database
Imports OneStream.Shared.Engine
Imports OneStream.Shared.Wcf
Imports OneStream.Stage.Database
Imports OneStream.Stage.Engine
Imports OneStreamWorkspacesApi
Imports OneStreamWorkspacesApi.V800

Namespace Workspace.__WsNamespacePrefix.__WsAssemblyName
	Public Class Migrations
		
		'Get all the table classes
		Dim tables As New List(Of Object) From {
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.raw_work_order,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.raw_work_order_figures,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.raw_structure_figures,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.raw_fxrate,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.dim_entity,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.dim_account,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_fxrate,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_serv_type,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_rev_type,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_entity_currency,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_scenario_fxratetype,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_department,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_technology,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_year_scenario_confirm,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.aux_template_warning,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.work_order,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.work_order_figures,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.structure_figures,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.function_translate_amount,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.view_work_order_figures_horizontal,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.view_work_order_figures_ytd,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.view_work_order_figures_translated,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.view_structure_figures_horizontal,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.view_structure_figures_ytd,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.view_structure_figures_translated,
			New Workspace.__WsNamespacePrefix.__WsAssemblyName.serv_allocations_main
		}

		#Region "Get Migration Queries"
		
        Public Function GetMigrationQueries(ByVal si As SessionInfo, ByVal type As String) As List(Of String)
            Try
				'Declare list of queries
				Dim queries As New List(Of String)
				
				'Loop through all the tables to get the queries
				For Each table In tables
					queries.Add(table.GetMigrationQuery(si, type))
				Next
				
                Return queries
            Catch ex As Exception
                Throw New XFException(si, ex)
			End Try
        End Function
		
		#End Region

		#Region "Get Population Queries"
		
        Public Function GetPopulationQueries(ByVal si As SessionInfo, ByVal type As String) As List(Of String)
            Try
				'Declare list of queries
				Dim queries As New List(Of String)
				
				'Loop through all the tables to get the queries
				For Each table In tables
					queries.Add(table.GetPopulationQuery(si, type))
				Next
				
                Return queries
            Catch ex As Exception
                Throw New XFException(si, ex)
			End Try
        End Function
		
		#End Region

	End Class
End Namespace
